README.md

```diff
@@ -619,7 +619,30 @@ Some pointers:
   key to the translation file and use the arrow function version, like so:
   `nameOfTheKey: ({amount, dateTime}) => "User has sent " + amount + " to you on " + dateTime,`.
   This is because the order of the phrases might vary from one language to another.
-
+- When working with translations that involve plural forms, it's important to handle different cases correctly.
+
+  For example:
+  - zero: Used when there are no items **(optional)**. 
+  - one: Used when there's exactly one item.
+  - two: Used when there's two items. **(optional)**
+  - few: Used for a small number of items **(optional)**.
+  - many: Used for larger quantities **(optional)**.
+  - other: A catch-all case for other counts or variations.
+
+  Here’s an example of how to implement plural translations:
+
+  messages: () => ({
+      zero: 'No messages',
+      one: 'One message',
+      two: 'Two messages',
+      few: (count) => `${count} messages`,
+      many: (count) => `You have ${count} messages`,
+      other: (count) => `You have ${count} unread messages`,
+  })
+
```

Comment:
> No, since it will be impossible to differentiate between messages and messages.one translation key in a flat object.

I don't get this

> This implementation looks like any other object with translations
> And not a specific translation with plural forms

Yeah, so?

Anyway, it would be nice if we supported the simple object notation too, but not going to block on that since this is a big PR, we've been waiting a long time for it and adding the function is not that bad.